File: src/block/GrassBlock.java
package block;

public class GrassBlock extends Block {
    public GrassBlock() {
        super("grass", new BlockAttribute("snowy", BlockProperty.BOOLEAN));
//        super("grass");
    }

    @Override
    protected boolean isFullCube(BlockState blockState) {
        return true;
    }

    @Override
    public BlockModel getModel(BlockState blockState) {
        return new BlockModel(new int[][] {
                {0x00000000, 0x0f0f0f00},
                {0x00000f01, 0x0f0f0f00},
                {0x00000002, 0x0f0f0f00},
                {0x0f000003, 0x0f0f0f00},
                {0x00000004, 0x0f0f0f00},
                {0x000f0005, 0x0f0f0f00},
        });
    }
}

File: java/src/block/GrassBlock.kt
package src.block

import src.world.Chunk.Norm.*

class GrassBlock : Block("grass", BlockAttribute("snowy", BlockProperty.BOOLEAN)) {
    override fun getModel(blockState: BlockState): BlockModel {
        return if (blockState.properties[0].toInt() == 0) BlockModel(arrayOf(
                BlockFace(SOUTH.ordinal,
                        x =  0, y =  0, z = 0,
                        w = 15, h = 15, t = 0),
                BlockFace(NORTH.ordinal,
                        x =  0, y =  0, z = 0,
                        w = 15, h = 15, t = 0),
        )) else BlockModel(arrayOf(
                BlockFace(SOUTH.ordinal,
                        x =  0, y =  0, z = 0,
                        w = 15, h = 15, t = 0),
                BlockFace(NORTH.ordinal,
                        x =  0, y =  0, z = 0,
                        w = 15, h = 15, t = 0),
                ))
    }

    override fun hasModel(blockState: BlockState): Boolean {
        return true
    }
}